### ISSUE TITLE
Saving a panel consisting of pn.Row or pn.Column elements is buggy.

### ISSUE THREAD

**Issue by locsta** [holoviz/panel#2377]
```
def plot_correlations(media, y="result"):
    
    # Create scatter plot and slope and assemble them together in the variable chart
    scatter = df.plot.scatter(**params)
 
    slope = hv2.Slope(gradient, y_intercept)                         
    chart = scatter * slope.opts(color="red", line_width=3)
                                  
    # DataFrame
    df_describe = pd.DataFrame([{"Col_1":"test1", "Col_2":"test2}])
    chart_panel = pn.Column(chart, pn.pane.DataFrame(df_describe))
    
    return chart_panel
```

```
Spends = ["a", "b", "c", "d", "e"]
kw = dict(media=sorted(list(Spends)))
total_corr = pn.interact(plot_correlations, **kw)
total_corr.save("test.html", embed=True)
```

The panel resulting looks great in Jupyter Notebooks and work as expected, but when I save the panel to html I get the following ouput:
  0%|          | 0/5 [00:00<?, ?it/s]
 20%|██        | 1/5 [00:00<00:01,  2.04it/s]
 40%|████      | 2/5 [00:00<00:01,  2.03it/s]
 60%|██████    | 3/5 [00:01<00:00,  2.15it/s]
 80%|████████  | 4/5 [00:01<00:00,  2.53it/s]

The last combination is not saved.
Basically every panel being a combination of panels created by pn.Column or pn.Row is not exported properly.
Everytime the last combination is not exported

**Comment by MarcSkovMadsen** [comment]
Hi @locsta . 

Thanks for reporting.

Could you please provide a minimum reproducible example? It simply makes it much easier to understand, reproduce and solve. 

Thanks,.

**Comment by jcorb** [comment]
Hi @MarcSkovMadsen - I'm not the original author, but having the same issue. Here is reproducible example

```
from bokeh.plotting import figure
from bokeh.models import ColumnDataSource
import panel as pn
import numpy as np
from bokeh.resources import CDN

def plot_data(data_id):

    source = sources[data_id]
    p1 = figure(plot_height=500, plot_width=500)
    p1.line('x', 'y', source=source, color='steelblue', line_width=3)
    
    return pn.Column(p1)


sources = {1:ColumnDataSource(data={'x': np.arange(0,100),
                                    'y': np.arange(0,100)**2}),
           2:ColumnDataSource(data={'x': np.arange(0,100),
                                    'y': np.arange(0,100)*3}),
           3:ColumnDataSource(data={'x': np.arange(0,100),
                                    'y': np.arange(0,100)*0.5})}

interact = pn.interact(plot_data, data_id=list(sources.keys()))

interactive_plot = pn.Column(interact[0], interact[1])

interactive_plot.save('panel_test.html', resources=CDN, embed=True, title='panel test')
# interactive_plot.show() works as expected
# interactive_plot.show() 
```

`panel_test.html`  displays a slider bar widget and a basic x/y plot below it. Changing the slider values from 1 to 2 changes the plot as expected, the problem arises when changing the slider from 2 to 3. The slider value changes, but no update of the plot occurs. When using the `.show()` method the interaction works as expected.


My env:
```
(panel-testing) josephcorbett@joe panel-test % conda list 
# packages in environment at /Users/josephcorbett/miniconda3/envs/panel-testing:
#
# Name                    Version                   Build  Channel
bokeh                     2.3.2            py38h50d1736_0    conda-forge
brotlipy                  0.7.0           py38h5406a74_1001    conda-forge
ca-certificates           2021.5.30            h033912b_0    conda-forge
certifi                   2021.5.30        py38h50d1736_0    conda-forge
cffi                      1.14.5           py38ha97d567_0    conda-forge
chardet                   4.0.0            py38h50d1736_1    conda-forge
cryptography              3.4.7            py38h1fa4640_0    conda-forge
freetype                  2.10.4               h4cff582_1    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
importlib-metadata        4.6.0            py38h50d1736_0    conda-forge
jbig                      2.1               h0d85af4_2003    conda-forge
jinja2                    3.0.1              pyhd8ed1ab_0    conda-forge
jpeg                      9d                   hbcb3906_0    conda-forge
lcms2                     2.12                 h577c468_0    conda-forge
lerc                      2.2.1                h046ec9c_0    conda-forge
libblas                   3.9.0                9_openblas    conda-forge
libcblas                  3.9.0                9_openblas    conda-forge
libcxx                    11.1.0               habf9029_0    conda-forge
libdeflate                1.7                  h35c211d_5    conda-forge
libffi                    3.3                  h046ec9c_2    conda-forge
libgfortran               5.0.0           9_3_0_h6c81a4c_22    conda-forge
libgfortran5              9.3.0               h6c81a4c_22    conda-forge
liblapack                 3.9.0                9_openblas    conda-forge
libopenblas               0.3.15          openmp_h5e1b9a4_1    conda-forge
libpng                    1.6.37               h7cec526_2    conda-forge
libtiff                   4.3.0                h1167814_1    conda-forge
libwebp-base              1.2.0                h0d85af4_2    conda-forge
llvm-openmp               11.1.0               hda6cdc1_1    conda-forge
lz4-c                     1.9.3                h046ec9c_0    conda-forge
markdown                  3.3.4              pyhd8ed1ab_0    conda-forge
markupsafe                2.0.1            py38h96a0964_0    conda-forge
ncurses                   6.2                  h2e338ed_4    conda-forge
numpy                     1.21.0           py38had91d27_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjpeg                  2.4.0                h6e7aa92_1    conda-forge
openssl                   1.1.1k               h0d85af4_0    conda-forge
packaging                 20.9               pyh44b312d_0    conda-forge
panel                     0.11.3             pyhd8ed1ab_0    conda-forge
param                     1.10.1             pyhd3deb0d_0    conda-forge
pillow                    8.2.0            py38h83525de_1    conda-forge
pip                       21.1.3             pyhd8ed1ab_0    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyct                      0.4.6                      py_0    conda-forge
pyct-core                 0.4.6                      py_0    conda-forge
pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pysocks                   1.7.1            py38h50d1736_3    conda-forge
python                    3.8.10          h0e5c897_0_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.8                      2_cp38    conda-forge
pyviz_comms               2.0.2              pyhd8ed1ab_0    conda-forge
pyyaml                    5.4.1            py38h5406a74_0    conda-forge
readline                  8.1                  h05e3726_0    conda-forge
requests                  2.25.1             pyhd3deb0d_0    conda-forge
setuptools                49.6.0           py38h50d1736_3    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.36.0               h23a322b_0    conda-forge
tk                        8.6.10               h0419947_1    conda-forge
tornado                   6.1              py38h5406a74_1    conda-forge
tqdm                      4.61.1             pyhd8ed1ab_0    conda-forge
typing_extensions         3.10.0.0           pyha770c72_0    conda-forge
urllib3                   1.26.6             pyhd8ed1ab_0    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
yaml                      0.2.5                haf1e3a3_0    conda-forge
zipp                      3.4.1              pyhd8ed1ab_0    conda-forge
zlib                      1.2.11            h7795811_1010    conda-forge
zstd                      1.5.0                h582d3a0_0    conda-forge
```

**Comment by jcorb** [comment]
A further note that it seems to have something to do with `pn.interact()`. When using a parameterised class approach it works as expected
```
from bokeh.plotting import figure
from bokeh.models import ColumnDataSource
import panel as pn
import param
import numpy as np
from bokeh.resources import CDN, INLINE


class dataPlot(param.Parameterized):
    sources = {1:ColumnDataSource(data={'x': np.arange(0,10),
                                    'y': np.arange(0,10)**2}),
           2:ColumnDataSource(data={'x': np.arange(0,10),
                                    'y': np.arange(0,10)*3}),
           3:ColumnDataSource(data={'x': np.arange(0,10),
                                    'y': np.arange(0,10)**0.5})}
    select = param.Selector(objects=list(sources.keys()))
    @param.depends('select')    
    def view(self):
        source = self.sources[self.select]
        p1 = figure(plot_height=500, plot_width=500)
        p1.line('x', 'y', source=source, color='steelblue', line_width=3)
    
        return pn.Column(p1)

data = dataPlot(name='Test')
layout = pn.Row(data.param, data.view)
# layout.show()
layout.save('param_test.html', resources=CDN, embed=True, title='param test')
```
